src/plugins/intel_cpu/src/emitters/plugin/riscv64/jit_eltwise_emitters.cpp

```diff
@@ -324,6 +324,89 @@ void jit_exp_emitter::register_table_entries() {
     push_arg_entry_of("exponent_bias", 0x0000007f);
 }
 
+/// MOD ///
+jit_mod_emitter::jit_mod_emitter(ov::intel_cpu::riscv64::jit_generator* host, ov::intel_cpu::riscv64::cpu_isa_t host_isa,
+                                 const std::shared_ptr<ov::Node>& node)
+    : jit_emitter(host, host_isa, get_arithmetic_binary_exec_precision(node)) {
+    prepare_table();
+}
+
+jit_mod_emitter::jit_mod_emitter(ov::intel_cpu::riscv64::jit_generator* host, ov::intel_cpu::riscv64::cpu_isa_t host_isa,
+                                 const ov::element::Type exec_prc)
+    : jit_emitter(host, host_isa, exec_prc) {
+    prepare_table();
+}
+
+size_t jit_mod_emitter::get_inputs_num() const {
+    return 2;
+}
+size_t jit_mod_emitter::aux_vecs_count() const {
+    if(exec_prc_ == ov::element::f32) {
+        return 1;
+    }
+    else if(exec_prc_ == ov::element::i32) {
+        return 0;
+    }
+    else {
+        OPENVINO_THROW("Unsupported precision");
+    }
+    
+}
+size_t jit_mod_emitter::aux_fp_gprs_count() const {
+    if(exec_prc_ == ov::element::f32) {
+        return 1;
+    }
+    else if(exec_prc_ == ov::element::i32) {
+        return 0;
+    }
+    else {
+        OPENVINO_THROW("Unsupported precision");
+    }
+}
+void jit_mod_emitter::emit_impl(const std::vector<size_t>& in_vec_idxs, const std::vector<size_t>& out_vec_idxs) const {
+    if (host_isa_ == ov::intel_cpu::riscv64::cpu_isa_t::gv) {
+        emit_isa<ov::intel_cpu::riscv64::cpu_isa_t::gv>(in_vec_idxs, out_vec_idxs);
+    } else {
+        OPENVINO_THROW("Can't create jit eltwise kernel");
+    }
+}
+
+template <ov::intel_cpu::riscv64::cpu_isa_t isa>
+void jit_mod_emitter::emit_isa(const std::vector<size_t>& in_vec_idxs, const std::vector<size_t>& out_vec_idxs) const {
+    VReg src0 = VReg(in_vec_idxs[0]);
+    VReg src1 = VReg(in_vec_idxs[1]);
+    VReg tmp1 = VReg(aux_vec_idxs[0]);
+    FReg fp0 = FReg(aux_fp_gpr_idxs[0]);
```

Comment:
Using the methods `aux_fp_gprs_count()` and `aux_vecs_count()`, we allocate needed registers only when execution precision is `fp32`. It means that it guarantees not empty `aux_fp_gpr_idxs` and `aux_vec_idxs` for `fp32`.

If the execution precision is `i32`, we don't guarantee that `aux_fp_gpr_idxs` and `aux_vec_idxs` will be allocated -> we cannot get access to the first elements here.

It means that we can call initialize `tmp1` and `fp0` registers only when the execution precision is `fp32`. Otherwise it might lead to segmentation fault :)

May I ask you please to fix this moment? 😊

Reply:
sure, I have fixed it, sorry i missed it 
Thank you